x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -435,6 +439,18 @@ public void testGetRolesForApiKey() throws Exception {
         }
     }
 
+    public void testApiKeyServiceDisabled() throws Exception {
+        final Settings settings = Settings.builder().put(XPackSettings.API_KEY_SERVICE_ENABLED_SETTING.getKey(), false).build();
+        final ApiKeyService service = createApiKeyService(settings);
+
+        ElasticsearchException e = expectThrows(ElasticsearchException.class,
+            () -> service.getApiKeys(randomAlphaOfLength(6), randomAlphaOfLength(8), null, null, new PlainActionFuture<>()));
+
+        assertThat(e, instanceOf(FeatureNotEnabledException.class));
+        assertThat(e, throwableWithMessage("api keys are not enabled"));
+        assertThat(e.getMetadata(FeatureNotEnabledException.DISABLED_FEATURE_METADATA), contains("api_keys"));
```

Comment:
nit: Maybe replace `"api_keys"` string literal with `API_KEY_SERVICE.getFeatureName()`(this needs adding the getFeatureName method as well). I generaly prefer to have less string literals.

Reply:
I did this intentionally because the _contract_ for the client is the string, not the enum.
That is, I want to test that this actually has `"api_keys"` as the feature description, because that is what Kibana needs to see.

Perhaps I can add a comment to that effect?

Reply:
That's a good point and an interesting one. I haven't thought about that. A comment could be useful. Otherwise, I could be the next guy who refactor the enum name and also change the test here to match the refactor without aware of breaking client expectation. Thanks